codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -178,15 +204,22 @@ public float windowUpdateRatio(Http2Stream stream) throws Http2Exception {
     @Override
     public void receiveFlowControlledFrame(ChannelHandlerContext ctx, Http2Stream stream, ByteBuf data,
             int padding, boolean endOfStream) throws Http2Exception {
+        this.ctx = checkNotNull(ctx, "ctx");
```

Comment:
Actually, the `consumeBytes` method isn't guaranteed to have been called before the stream is removed. The logic for putting it here is that if there is any data to be consumed, then it must have gone through this method.